formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/aws/S3Client.scala

```diff
@@ -64,4 +69,43 @@ object S3 extends S3Client with LazyLogging {
       failedResults <- listFolderContents(config.resultsBucket, failedPath)
     } yield s3StatusResponse(failedResults, completedResults, config.resultsBucket)
   }
+
+  private def writeToS3(resultsBucket: String, filePath: String, contents: String): Either[Throwable, S3WriteSuccess] = Try {
+    s3Client.putObject(resultsBucket, filePath, contents)
+    s3Client.setObjectAcl(resultsBucket, filePath, CannedAccessControlList.BucketOwnerRead)
+  }.toEither.map(_ => S3WriteSuccess())
+
+  private def formatResults(results: List[FormstackSubmissionQuestionAnswer]): String = {
+    results.map { submission =>
+      val meta = List(s"Submission: ${submission.id}", s"Timestamp: ${submission.timestamp}")
+      val responses = submission.fields.map(field => s"${field.label}: ${field.value}")
+      (meta ::: responses).mkString("\n")
+    }.mkString("\n-------------------------\n")
+  }
+
+  override def writeSuccessResult(
+    initiationId: String,
+    results: List[FormstackSubmissionQuestionAnswer],
+    config: PerformSarLambdaConfig): Either[Throwable, S3WriteSuccess] = {
+    val resultsPath = s"${config.resultsPath}/$initiationId/completed"
+    if (results.nonEmpty) {
```

Comment:
i think we could replace this block with: 


```
 val filePath, contents =  if (results.nonEmpty) {
       logger.info("Writing SAR result to s3.")
       val filePath = s"$resultsPath/formstackSarResponse"
      val contents = formatResults(results)
      filePath, contents 
    } else {
      logger.info(s"No results found for request $initiationId. Creating NoResultsFoundForUser object.")
      val filePath = s"$resultsPath/noResultsFoundForUser"
      filePath, ""
    }

      writeToS3(config.resultsBucket, filePath, contents)
```

or better still if formatResults return empty string on the list being empty. so if condition check now return just the filepath and we could use writeToS3 as 
` 
writeToS3(config.resultsBucket, filePath, formatResults(contents))
 `

Reply:
Ah nice - thank you.